src/libstd/io/buffered.rs

```diff
@@ -451,7 +453,14 @@ impl<W: Write> BufWriter<W> {
     pub fn into_inner(mut self) -> Result<W, IntoInnerError<BufWriter<W>>> {
         match self.flush_buf() {
             Err(e) => Err(IntoInnerError(self, e)),
-            Ok(()) => Ok(self.inner.take().unwrap())
+            Ok(()) => unsafe {
+                // use `ptr::read` to extract the inner writer and circumvent
+                // our `Drop` implementation
+                mem::replace(&mut self.buf, Vec::new());
```

Comment:
Nevermind, I conflated it with `read_and_drop`.

On Thu, Sep 8, 2016 at 4:37 PM, bluss <notifications@github.com> wrote:

> In src/libstd/io/buffered.rs
> <https://github.com/rust-lang/rust/pull/36336#discussion_r78084035>:
>
> > @@ -451,7 +453,14 @@ impl<W: Write> BufWriter<W> {
> >      pub fn into_inner(mut self) -> Result<W, IntoInnerError<BufWriter<W>>> {
> >          match self.flush_buf() {
> >              Err(e) => Err(IntoInnerError(self, e)),
> > -            Ok(()) => Ok(self.inner.take().unwrap())
> > +            Ok(()) => unsafe {
> > +                // use `ptr::read` to extract the inner writer and circumvent
> > +                // our `Drop` implementation
> > +                mem::replace(&mut self.buf, Vec::new());
>
> Sure it works
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/rust-lang/rust/pull/36336/files/ea72b7efc9f8a7fc2234264cbc6fc2241f66cd52#r78084035>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAC3nxeWhhJT2ZkrrBbzzvR9e2an8Wp3ks5qoHHzgaJpZM4J3ebB>
> .
>